### ISSUE TITLE
progressbar should have aria-label/labelledby attribute set

### ISSUE THREAD

**Issue by jgiardino** [patternfly/patternfly#2539]
According to the accessibility tool available in Firefox 71, elements with `role="progressbar"` should have an accessible label: 

![image](https://user-images.githubusercontent.com/21063328/70946445-30090e00-2025-11ea-8f32-8169515f0c4e.png)

In a quick test in VO, there seemed to be no difference when replacing `aria-describedby` with `aria-labelledby`. However, given limited support for aria-describedby in other screen readers, it's possible that this change would be beneficial.

This attribute change should be properly tested, and verified that this switch makes sense with all our component examples to confirm that switching out these attributes is the right approach. It would also be worth considering making the visible text hidden from screen readers with `aria-hidden` if the label is provided on the element with `role="progressbar"` to avoid redundant text. 

This was found as part of a design review in the Cockpit team.

We could also include Spinner in this update which would require the following:
- aria-valuetext="Loading..." should become aria-label="Loading..."
- add aria-valuemin="0" and aria-valuemax="100" per recommendations from WAI-ARIA for indeterminate progressbar elements

**Comment by andreasn** [comment]
Related Mozilla docs: https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Text_labels_and_names?utm_source=devtools&utm_medium=a11y-panel-checks-text-label#Form_elements_must_be_labeled

**Comment by jgiardino** [comment]
Thanks for sharing that link, @andreasn! It helps to have an additional source to justify making this change. 

I pulled the following comment from this link:

> All elements within a form must have a `<label>` that identifies its purpose. This applies to all types of `<input>` items, as well as `<button>`, `<output>`, `<select>`, `<textarea>`, **`<progress>`** and `<meter>` elements, as well as any element with the switch ARIA role.

And also want to add that while we are not using `<progress>` for the Progress component, this element has an implicit role of `progressbar` according to the [Implicit ARIA roles of HTML elements](https://www.w3.org/TR/html-aria/) spec, which _is_ used for the Progress component. Assuming that any requirements noted for a semantic html element are also expected to be applied to the implicit role of that element, such as in the case of custom html elements that provide the same semantics as the semantic html element, then the requirements should also be applied when using that role. 

In other words, `<progress>` and `<div role="progressbar>` should be considered equal and treated the same regarding a11y requirements.
